### PR TITLE
[Enhancement] (nereids)implement showReplicaDistributionCommand in nereids

### LINKED ISSUE #42782: [Enhancement] (nereids)implement showREplicaDistributionCommand in nereids
showREplicaDistribution command is supported in legacy planner but not in nereids(new doris fe planner). you need implement the following showREplicaDistribution command in nereids:
unsupportedshowREplicaDistributionStatement
``` SHOW REPLICA DISTRIBUTION FROM baseTableRef                                   ```

### PR DESCRIPTION
**Pull request by Vallishp**
### What problem does this PR solve?

Issue Number: close #42782 
#42837 

Related PR: #xxx

Problem Summary:
implement showReplicaDistributionCommand in nereids
### Release note

None

### Check List (For Author)

- Test 
    - [X] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/TableRefInfo.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    : multipartIdentifier optScanParams? tableSnapshot? specifiedPartition?
+        tabletList? tableAlias sample? relationHint?
+    currently  tableNameInfo(multipartIdentifier) and specifiedPartition API exposed.
+    if you need write one and use.
+ */
+public class TableRefInfo {
+    private final TableNameInfo tableNameInfo;
+    private final TableScanParams scanParams;
+    private final TableSnapshot tableSnapShot;
+    private final PartitionNames partitionNames;
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/TableRefInfo.java:41]: You can use PartitionNamesInfo.java from https://github.com/apache/doris/pull/44058/files . This pr has some extra work to do and will not be merged in a short time. So you can add  PartitionNamesInfo.java in your pr.

**Comment by Vallishp** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowReplicaDistributionCommand.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import org.apache.doris.qe.StmtExecutor;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * show replica distribution command
+ */
+public class ShowReplicaDistributionCommand extends ShowCommand {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowReplicaDistributionCommand.java:47]: please help update the two functions in Command.java (add ConnectContext as param)
```
public void verifyCommandSupported(ConnectContext ctx) throws DdlException {
    // check command has been supported in cloud mode
    if (Config.isCloudMode()) {
        checkSupportedInCloudMode(ctx);
    }
}
```
`
protected void checkSupportedInCloudMode(ConnectContext ctx) throws DdlException {}`

and then change the caller in StmtExecutor.java
`((Command) logicalPlan).verifyCommandSupported(context);`

at last, override checkSupportedInCloudMode method as bellow
```
protected void checkSupportedInCloudMode(ConnectContext ctx) throws DdlException {
    if (!ctx.getCurrentUserIdentity().getUser().equals(Auth.ROOT_USER)) {
        LOG.info("stmt={}, not supported in cloud mode", stmt.toString());
        throw new AnalysisException("Unsupported operation");
    }
}
```

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**